From 20aae412f824d5245c8e2520b2e38713a64c73b5 Mon Sep 17 00:00:00 2001
From: LIU Hao <lh_mouse@126.com>
Date: Thu, 15 May 2025 19:12:51 +0800
Subject: [PATCH] Enable mcf thread model for aarch64-*-mingw*.

This is similar to d6d7afcdbc04adb0ec42a44b2d7e05600945af42 about the posix
and win32 thread model.

Signed-off-by: LIU Hao <lh_mouse@126.com>
Signed-off-by: Jonathan Yong <10walls@gmail.com>

libgcc/ChangeLog:

	* config.host: Enable mcf thread model for aarch64-*-mingw*.
	* config/i386/t-mingw-mcfgthread: Move to...
	* config/mingw/t-mingw-mcfgthread: ...here.
---
 libgcc/config.host                               | 7 +++++--
 libgcc/config/{i386 => mingw}/t-mingw-mcfgthread | 0
 2 files changed, 5 insertions(+), 2 deletions(-)
 rename libgcc/config/{i386 => mingw}/t-mingw-mcfgthread (100%)

diff --git a/libgcc/config.host b/libgcc/config.host
index 6a88ee5a2dd0..d36f0e34a3b6 100644
--- a/libgcc/config.host
+++ b/libgcc/config.host
@@ -472,6 +472,9 @@ aarch64-*-mingw*)
 	  posix)
 	    tmake_thr_file="mingw/t-mingw-pthread"
 	    ;;
+	  mcf)
+	    tmake_thr_file="mingw/t-mingw-mcfgthread"
+	    ;;
 	esac
 	tmake_file="${tmake_file} ${cpu_type}/t-no-eh ${tmake_thr_file}"
 	tmake_file="${tmake_file} t-dfprules"
@@ -904,7 +907,7 @@ i[34567]86-*-mingw*)
 	    tmake_thr_file="mingw/t-mingw-pthread"
 	    ;;
 	  mcf)
-	    tmake_thr_file="i386/t-mingw-mcfgthread"
+	    tmake_thr_file="mingw/t-mingw-mcfgthread"
 	    ;;
 	esac
 	# This has to match the logic for DWARF2_UNWIND_INFO in gcc/config/i386/cygming.h
@@ -931,7 +934,7 @@ x86_64-*-mingw*)
 	    tmake_thr_file="mingw/t-mingw-pthread"
 	    ;;
 	  mcf)
-	    tmake_thr_file="i386/t-mingw-mcfgthread"
+	    tmake_thr_file="mingw/t-mingw-mcfgthread"
 	    ;;
 	esac
 	# This has to match the logic for DWARF2_UNWIND_INFO in gcc/config/i386/cygming.h
diff --git a/libgcc/config/i386/t-mingw-mcfgthread b/libgcc/config/mingw/t-mingw-mcfgthread
similarity index 100%
rename from libgcc/config/i386/t-mingw-mcfgthread
rename to libgcc/config/mingw/t-mingw-mcfgthread
-- 
2.49.0

